src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -503,32 +509,54 @@ public async Task<List<PlatformSummary>> GetElasticBeanstalkPlatformArns()
                         Operator = "=",
                         Type = "PlatformStatus",
                         Values = { "Ready" }
+                        },
+                        new PlatformFilter
+                        {
+                            Operator = "contains",
+                            Type = "PlatformName",
+                            Values = { platformName }
                     }
                 }
             };
-            var response = await HandleException(async () => await beanstalkClient.ListPlatformVersionsAsync(request));
+
+                var platforms = await HandleException(async () => (await beanstalkClient.Paginators.ListPlatformVersions(request).PlatformSummaryList.ToListAsync()));
+
+                // Filter out old test platforms that only internal accounts would be able to see.
+                platforms = platforms.Where(x => !string.IsNullOrEmpty(x.PlatformBranchName)).ToList();
+
+                return platforms;
+            };
+
+            var allPlatformSummaries = new List<PlatformSummary>();
+            if (platformTypes.Contains(BeanstalkPlatformType.Linux))
+            {
+                allPlatformSummaries.AddRange(await fetchPlatforms(".NET Core"));
+            }
+            else if (platformTypes.Contains(BeanstalkPlatformType.Windows))
+            {
+                var windowsPlatforms = await fetchPlatforms("Windows Server");
+                SortElasticBeanstalkWindowsPlatforms(windowsPlatforms);
```

Comment:
I see where you are coming from but since we have the `GetLatestElasticBeanstalkPlatformArn` method in the `IAWSResourceQueryer` that implies we must be looking at the platforms in a sorted ordered. If we remove the sort method we would have to remove the `GetLatestElasticBeanstalkPlatformArn`. Maybe that is valid to do but too big of a change to make at this point.